app/routes/api/v1/ecommerce/orders/index.ts

```diff
@@ -0,0 +1,71 @@
+import type { ActionFunction } from "@remix-run/node";
+import { z } from "zod";
+import {
+  badRequestResponse,
+  methodNotAllowedResponse,
+  notFoundResponse,
+  okResponse,
+} from "~/helpers/response-helpers.server";
+import { parseBody } from "~/lib/parse-body.server";
+import { verifyCustomerRequest } from "~/lib/verify-request.server";
+import { createOrder } from "~/models/order/order.server";
+import {
+  findManyProductInstancesAvailableToOrder,
+  updateManyProductInstanceOrders,
+} from "~/models/productInstance/productInstance.server";
+
+export const action: ActionFunction = async ({
+  request,
+}): Promise<Response> => {
+  switch (request.method.toLowerCase()) {
+    case "post": {
+      verifyCustomerRequest(request);
+      return handlePOSTRequest(request);
+    }
+
+    default: {
+      return methodNotAllowedResponse();
+    }
+  }
+};
```

Comment:
If this route will only accept POST requests, I don't see any use of the switch here.

The same could be achieved with a `if-else`.

However, if the route will accept in the future others requests, could be easily be added with the `switch`.